javascript/ql/src/semmle/javascript/security/performance/PolynomialReDoSCustomizations.qll

```diff
@@ -70,14 +83,22 @@ module PolynomialReDoS {
     }
 
     override RegExpTerm getRegExp() { result = term }
+
+    override DataFlow::Node getHighlight() { result = mcn }
   }
 
   /**
    * An operation that limits the length of a string, seen as a sanitizer.
    */
   class StringLengthLimiter extends Sanitizer {
     StringLengthLimiter() {
-      this.(StringReplaceCall).isGlobal()
+      this.(StringReplaceCall).isGlobal() and
+      // not char classes - they don't remove any repeated pattern.
+      not exists(RegExpTerm root | root = this.(StringReplaceCall).getRegExp().getRoot() |
+        root instanceof RegExpCharacterClass
+        or
+        root instanceof RegExpCharacterClassEscape
+      )
```

Comment:
Two stray thoughts that can be handled separately.

- should we have `RegExpTerm::getUnderlyingTerm` to easily handle the `/((([a-z])))/g` case as well?
- should we have a unifying class for the `RegExpCharacterClass` and `RegExpCharacterClassEscape` classes?

Reply:
> should we have RegExpTerm::getUnderlyingTerm to easily handle the /((([a-z])))/g case as well?   

We could - it sounds like an easy addition (for another PR).  
In that case we should also look around for other places where such a predicate could be useful. 

> should we have a unifying class for the RegExpCharacterClass and RegExpCharacterClassEscape classes?

I don't think so. The naming will probably just get even more confusing.   
But I'm open to naming suggestions.

Reply:
I have created two issues.